packages/edit-site/src/store/reducer.js

```diff
@@ -139,12 +139,7 @@ export function templateType( state = 'wp_template', action ) {
 export function page( state = {}, action ) {
 	switch ( action.type ) {
 		case 'SET_PAGE':
-			return { contentInfo: action.page, templateId: action.templateId };
-		case 'SET_TEMPLATE':
-			if ( action.templateId === state.templateId ) {
-				return state;
-			}
```

Comment:
I think these changes regarding `SET_TEMPLATE` and the page state structure are still necessary, however they fixed a bug that is present regardless of our query arg changes here.  Consider the flow where:

* A user sets a page to edit.  The editor loads the page and corresponding template.
* The user is done editing the page and wants to go edit a different template.  The user navigates to this template and selects it in the nav sidebar.
* Without this check, the page context will still be set for that unrelated template.

Specifically you can see this bug now if you load the editor (loads on front-page template), then select the sample page, then select the front-page template again you will see the sample page's content in the front-page's post content block.

This shouldn't block the PR though, we can reintroduce this bug fix separately or in the follow up for the taxonomy.

Reply:
I'm ok following up on it. I wasn't sure exactly how to handle this situation. I could see it being good to keep the page context around -- then you can preview it in different templates. But I can see how that could also be a bug 🤔

Reply:
Yeah it is a bit confusing in that regard.  I would argue that switching back to a specific template should clear the page context b/c at that point our context is template editing.  

> I could see it being good to keep the page context around -- then you can preview it in different templates. 

I think we need a separate new mechanism for changing / previewing a different template for the currently selected content.  This current flow is definitely a hacky way to preview that stems from outdated state and doesn't allow the user to actually set anything.

Reply:
Yeah, that's a good point. I really would like to avoid putting the template ID in the page state. I guess it doesn't _really_ matter, but it seems weird to have to store it twice

Reply:
>  it seems weird to have to store it twice

True.  I guess one corresponds to which template is loaded in the editor, while the other corresponds to which is resolved for the page/content.  Maybe the latter should added to the page object itself as `resolvedTempalate` or something similar.  Or maybe there is a better way to handle this.

Reply:
I like the resolved template idea!

Reply:
We will have to look into that.  Anyways, il mark this convo as 'resolved' since I don't want to hold up this PR on an unrelated tangent conversation. 🤣